Source/JavaScriptCore/assembler/MacroAssemblerX86Common.cpp

```diff
@@ -802,30 +802,31 @@ void MacroAssemblerX86Common::collectCPUFeatures()
 {
     static std::once_flag onceKey;
     std::call_once(onceKey, [] {
+#if OS(DARWIN)
+        uint64_t capabilities = *reinterpret_cast<volatile const uint64_t*>(_COMM_PAGE_CPU_CAPABILITIES64);
+        s_sse4_1CheckState = (capabilities & kHasSSE4_1) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+        s_sse4_2CheckState = (capabilities & kHasSSE4_2) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+        s_popcntCheckState = (capabilities & kHasSSE4_2) ? CPUIDCheckState::Set : CPUIDCheckState::Clear; // POPCNT is part of SSE4.2.
+        s_avxCheckState = (capabilities & kHasAVX1_0) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+        s_bmi1CheckState = (capabilities & kHasBMI1) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+        s_lzcntCheckState = (capabilities & kHasBMI1) ? CPUIDCheckState::Set : CPUIDCheckState::Clear; // LZCNT is part of BMI1.
```

Comment:
Comes from https://github.com/apple/darwin-xnu/blob/main/osfmk/i386/cpu_capabilities.h

Reply:
this is nice